index.bs

```diff
@@ -8083,21 +8079,395 @@ objects.
                 return <emu-val>undefined</emu-val>.
             1.  Otherwise, return the result of performing any steps that were required to be run if the promise was fulfilled,
                 with |value| as the promise’s value.
-    1.  Let |onRejected| be a new [=built-in function object=] whose
-        behavior when invoked is as follows:
+    1.  Let |onFulfilled| be [=!=] [$CreateBuiltinFunction$](|onFulfilledSteps|, « »):
+    1.  Let |onRejectedSteps| be the following steps:
         1.  Let |R| be the first argument to |onRejected|.
         1.  Let |reason| be the result of [=converted to an IDL value|converting=]
             |R| to an IDL value of type {{any}}.
         1.  If there are no steps that are required to be run if the promise was rejected, then
             return <emu-val>undefined</emu-val>.
         1.  Otherwise, return the result of performing any steps that were required to be run if the promise was rejected,
             with |reason| as the rejection reason.
-    1.  Return [=!=] [$PerformPromiseThen$](|promise|, |onFulfilled|, |onRejected|).
+    1.  Let |onRejected| be [=!=] [$CreateBuiltinFunction$](|onRejectedSteps|, « »):
+    1.  Let |constructor| be |promise|.\[[Promise]].\[[Realm]].\[[Intrinsics]].[[{{%Promise%}}]].
+    1.  Let |newCapability| be ? [$NewPromiseCapability$](|constructor|).
+    1.  Return [=!=] [$PerformPromiseThen$](|promise|.\[[Promise]], |onFulfilled|, |onRejected|, |newCapability|).
```

Comment:
You don't need to pass newCapability since it's unused. Just pass three arguments.

Reply:
Well, current "upon fulfillment" etc. usages don't have a return value... that's where you'd use "transforming". But I see you seem to be allowing this to stand in for both, hmm.

Reply:
I guess we can just update all callers to say "Let _x_ be the result of performing the following steps upon fulfillment of _p_, given _v_" instead of the current "Let _x_ be the result of transforming _p_ by the following fulfillment steps, given _v_". That's not very different.

Reply:
Right, I meant to call that out explicitly, but it seems I forgot. I could add a note that it's not necessary to create the PromiseCapability